deploy/index-image/bundle.Dockerfile.ci-index-image-upgrade

```diff
@@ -0,0 +1,28 @@
+FROM quay.io/fedora/fedora:33-x86_64 AS builder
+ARG INITIAL_VERSION=1.4.0
+ARG TARGET_VERSION=100.0.0
```

Comment:
Not sure why we need a builder image here, just to sedding two files. I think it can be done within one Dockerfile, or from the outside before calling to docker build; e.g. copy all the files to a temporary directory, do the change and then call to docker build.

Reply:
Also, I find the file name a bit confusing: I think it should start with `Dockerfile` and the rest will be its postfix. Using `Dockerfile` as postfix will also work (I not sure I like it better),

Reply:
> Not sure why we need a builder image here, just to sedding two files. I think it can be done within one Dockerfile, or from the outside before calling to docker build; e.g. copy all the files to a temporary directory, do the change and then call to docker build.

After reading openshift/release#17442, I understand why it should be done within a Dockerfile, but I still think that the builder pattern is not needed here. However, I would join all the `RUN` commands into one, in order to create less layers; e.g.
```
COPY ${PACKAGE_NAME}/${INITIAL_VERSION}/manifests/ /manifests/
COPY ${PACKAGE_NAME}/${INITIAL_VERSION}/metadata /metadata/
RUN mv /manifests/kubevirt-hyperconverged-operator.v${INITIAL_VERSION}.clusterserviceversion.yaml \
    /manifests/kubevirt-hyperconverged-operator.v${TARGET_VERSION}.clusterserviceversion.yaml && \
  sed -i 's/${INITIAL_VERSION}/${TARGET_VERSION}/g' /manifests/kubevirt-hyperconverged-operator.v${TARGET_VERSION}.clusterserviceversion.yaml && \
  sed -i 's/${INITIAL_VERSION}/${TARGET_VERSION}/g' /metadata/annotations.yaml
```

Reply:
> Not sure why we need a builder image here, just to sedding two files. I think it can be done within one Dockerfile, or from the outside before calling to docker build; e.g. copy all the files to a temporary directory, do the change and then call to docker build.

There is no way to do the modification on the ci-operator config, and I wanted to avoid having 100.0.0 manifests at the directory. Also, I didn't find a way to resolve the `INITIAL_VERSION` (currently 1.4.0) in the dockerfile context, that's why I used an additional ARG.



> Also, I find the file name a bit confusing: I think it should start with `Dockerfile` and the rest will be its postfix. Using `Dockerfile` as postfix will also work (I not sure I like it better),

I'll update the dockerfile name.



> > Not sure why we need a builder image here, just to sedding two files. I think it can be done within one Dockerfile, or from the outside before calling to docker build; e.g. copy all the files to a temporary directory, do the change and then call to docker build.
> 
> After reading [openshift/release#17442](https://github.com/openshift/release/pull/17442), I understand why it should be done within a Dockerfile, but I still think that the builder pattern is not needed here. However, I would join all the `RUN` commands into one, in order to create less layers; e.g.
> 
> ```
> COPY ${PACKAGE_NAME}/${INITIAL_VERSION}/manifests/ /manifests/
> COPY ${PACKAGE_NAME}/${INITIAL_VERSION}/metadata /metadata/
> RUN mv /manifests/kubevirt-hyperconverged-operator.v${INITIAL_VERSION}.clusterserviceversion.yaml \
>     /manifests/kubevirt-hyperconverged-operator.v${TARGET_VERSION}.clusterserviceversion.yaml && \
>   sed -i 's/${INITIAL_VERSION}/${TARGET_VERSION}/g' /manifests/kubevirt-hyperconverged-operator.v${TARGET_VERSION}.clusterserviceversion.yaml && \
>   sed -i 's/${INITIAL_VERSION}/${TARGET_VERSION}/g' /metadata/annotations.yaml
> ```

will do.